sklearn/base.py

```diff
@@ -296,6 +297,24 @@ def __repr__(self):
         return '%s(%s)' % (class_name, _pprint(self.get_params(deep=False),
                                                offset=len(class_name),),)
 
+    def __getstate__(self):
+        if type(self).__module__.startswith('sklearn.'):
+            return dict(self.__dict__.items(), __version__=__version__)
+        else:
+            return dict(self.__dict__.items())
+
+    def __setstate__(self, state):
+        if type(self).__module__.startswith('sklearn.'):
+            pickle_version = state.pop("__version__", "pre-0.18")
```

Comment:
I am afraid that `__version__` is very generic attribute name and could potentially conflict with other attribute with the same name either a downstream project or with future Python.

I would would feel more comfortable to name this: `_sklearn_version` and let leave the double-sided `__` attribute name convention to Python language-level features.